src/main/java/com/diffmin/Main.java

```diff
@@ -16,12 +21,23 @@
      * @param newFile Modified version of the file
      * @throws FileNotFoundException Exception is raised when path of either file is invalid
      */
-    public static CtModel patchAndGenerateModel(File prevFile, File newFile)
-            throws FileNotFoundException {
-        App app = new App();
+    static CtModel patchAndGenerateModel(File prevFile, File newFile) throws FileNotFoundException {
         Pair<Diff, CtModel> diffAndModel = App.computeDiff(prevFile, newFile);
-        app.generatePatch(diffAndModel.getFirst());
-        app.applyPatch();
+
+        // Generate patches
+        Generation patchGenerationFactory = new Generation();
```

Comment:
This is not a factory in the sense that the word is most often used, see [the factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern). That could be confusing.